spec/unit/util_spec.rb

```diff
@@ -871,13 +871,13 @@ def withenv_utf8(&block)
       end
     end
 
-    it "should copy the permissions of the source file before yielding on Unix", :if => !Puppet.features.microsoft_windows? do
```

Comment:
To satisfy the criteria in PUP-7394, I think we also need a test that works like the example mentioned in the ticket:

```ruby
Puppet::Util.replace_file('/tmp/foo', 0440) { |of| FileUtils.copy_stream(StringIO.new("abcdef"), of) }
```

We should pass in a mode that doesn't allow writes, try to write... and verify the mode on the file is correct *and* the contents are as expected.

Reply:
Yeah, that's being indirectly tested here due to how the implementation of `replace_file` works, but it would be good to have that scenario explicitly tested in a separate test.